### PR TITLE
#168363049 Get device API endpoint

### PR DESCRIPTION
**Pull request by okiti**
**What does this PR do?**

1. get single device using the device _id
2. Validate token using an Auth middleware
3. Tests the get  device endpoint
**What are the relevant pivotal tracker stories?**
[#168363049](https://www.pivotaltracker.com/story/show/168363049)

### CODE REVIEW

File: src/http/middleware/Auth.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import jwt from 'jsonwebtoken';
+import dotenv from 'dotenv';
+import db from '../../db';
+
+dotenv.config();
+
+// import/prefer-default-export
+const Auth = async (req, res, next) => {
```
Comment (IAMOTZ) [inline src/http/middleware/Auth.js]: Any reason you decided to capitalize `Auth`?

File: src/http/validations/device.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+const getSingleDevice = (req, res, next) => {
+  const { deviceId } = req.params;
+
+  const fieldErrors = new FieldErrors();
+
+  if (!deviceId || typeof (deviceId) !== 'string') fieldErrors.addError('deviceId', 'deviceId is a required string');
+
+  if (fieldErrors.count > 0) {
+    return res.status(400).json({ success: false, message: 'Invalid request body', errors: fieldErrors.errors });
```
Comment (IAMOTZ) [inline src/http/validations/device.js]: `deviceId` is not in the request body so I think a message like this wold be better:
```suggestion
    return res.status(400).json({ success: false, message: 'Invalid request', errors: fieldErrors.errors });
```

File: src/db/device/deviceModel.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import mongoose from 'mongoose';
+
+const { Schema } = mongoose;
+
+const deviceSchema = new Schema({
+  hospitalId: { type: Schema.Types.ObjectId, required: true },
+  wardId: { type: Schema.Types.ObjectId },
+  label: { type: String, required: true },
```
[the remark this answers is not in the document]
Comment (okiti) [reply src/db/device/deviceModel.js]: fixed

File: src/db/device/deviceModel.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+  dateCreated: { type: Date, default: Date.now(), required: true },
```
[the remark this answers is not in the document]
Comment (okiti) [reply src/db/device/deviceModel.js]: fixed

File: src/db/device/deviceModel.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+});
+
+export default {
+  deviceModel: mongoose.model('devices', deviceSchema),
+};
```
[the remark this answers is not in the document]
Comment (okiti) [reply src/db/device/deviceModel.js]: fixed

File: src/db/device/deviceModel.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+
+const deviceSchema = new Schema({
+  hospitalId: { type: Schema.Types.ObjectId, required: true },
+  wardId: { type: Schema.Types.ObjectId },
+  label: { type: String, required: true },
+  dateCreated: { type: Date, default: Date.now(), required: true },
+});
+
+export default {
+  deviceModel: mongoose.model('devices', deviceSchema),
```
[the remark this answers is not in the document]
Comment (okiti) [reply src/db/device/deviceModel.js]: fixed

File: src/db/device/index.js [hunk 7]
```diff
@@ -0,0 +1,17 @@
+import deviceSchema from './deviceModel';
```
[the remark this answers is not in the document]
Comment (okiti) [reply src/db/device/index.js]: fixed

File: src/http/middleware/Auth.js [hunk 8]
```diff
@@ -0,0 +1,54 @@
+import jwt from 'jsonwebtoken';
+import dotenv from 'dotenv';
+import db from '../../db';
+
+dotenv.config();
+
+// import/prefer-default-export
```
[the remark this answers is not in the document]
Comment (okiti) [reply src/http/middleware/Auth.js]: fixed

File: src/http/middleware/Auth.js [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+      nurse_user: {
+        ...basic,
+        hospitalId: user.hospitalId,
+        wardId: user.wardId,
+      },
+    };
+
+    res.locals.user = userProfile[type];
+  } catch (err) {
+    return res.status(403).json({ success: false, message: `${err.name}: ${err.message}` });
```
[the remark this answers is not in the document]
Comment (okiti) [reply src/http/middleware/Auth.js]: fixed

File: src/http/middleware/Auth.js [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+      ward_user: {
+        ...basic,
+        hospitalId: user.hospitalId,
+      },
+      nurse_user: {
+        ...basic,
+        hospitalId: user.hospitalId,
+        wardId: user.wardId,
+      },
+    };
```
[the remark this answers is not in the document]
Comment (okiti) [reply src/http/middleware/Auth.js]: fixed

File: src/http/middleware/Auth.js (hunk 1 above)
Comment (okiti) [reply src/http/middleware/Auth.js]: fixed

File: src/http/validations/device.js (hunk 2 above)
Comment (okiti) [reply src/http/validations/device.js]: fixed

**Pull request merged**